includes/commerce_pos.payment.inc

```diff
@@ -40,11 +48,12 @@ function commerce_pos_payment(&$form, &$form_state, $transaction_type) {
 
   form_load_include($form_state, 'inc', 'commerce_pos', 'includes/commerce_pos.common');
   commerce_pos_payment_ajax_check($form, $form_state);
+  dpm($form_state);
 
   $return = $transaction_type == CommercePosService::TRANSACTION_TYPE_RETURN;
   $payment_options = commerce_pos_get_payment_options();
   $form['#theme'] = 'commerce_pos_payment';
-  $form['#redirect_key'] = $return ? 'return' : 'sale';
+  $form['#redirect_key'] = $return ? 'return' : ($transaction_type == CommercePosService::TRANSACTION_TYPE_EXCHANGE ? 'exchange' : 'sale');
```

Comment:
Might be better to break this out into a switch statement for readability. There are 3 types of these transaction types now that we are checking for. Or, to make things easier, since the redirect key is going to be either 'return', 'sale', or 'exchange', and the $transaction_type is already one of those three, you can just set $form['#redirect_key'] = $transaction_type